frictionless/detector.py

```diff
@@ -122,48 +124,181 @@ def __setattr__(self, name, value):
 
     @property
     def buffer_size(self):
```

Comment:
We can also declare an ouput type here and for other properties using Python Typing